drivers/gpio/gpio_mcp23s08.c

```diff
@@ -0,0 +1,452 @@
+					uint32_t mask)
+{
+	return mcp23s08_port_set_masked_raw(dev, mask, 0);
+}
+
+static int mcp23s08_port_toggle_bits(const struct device *dev, uint32_t mask)
+{
+	struct mcp23s08_drv_data *const drv_data =
+		(struct mcp23s08_drv_data *const)dev->data;
+	uint8_t buf;
+	int ret;
+
+	/* Can't do SPI bus operations from an ISR */
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	k_sem_take(&drv_data->lock, K_FOREVER);
+
+	buf = drv_data->reg_cache.gpio;
+	buf ^= mask;
+
+	ret = write_port_regs(dev, REG_GPIO, buf);
+	if (ret == 0) {
+		drv_data->reg_cache.gpio = buf;
+	}
+
+	k_sem_give(&drv_data->lock);
+
+	return ret;
+}
+
+static int mcp23s08_pin_interrupt_configure(const struct device *dev,
+					    gpio_pin_t pin,
+					    enum gpio_int_mode mode,
+					    enum gpio_int_trig trig)
+{
+	return -ENOTSUP;
+}
+
+static const struct gpio_driver_api api_table = {
+	.pin_configure = mcp23s08_config,
+	.port_get_raw = mcp23s08_port_get_raw,
+	.port_set_masked_raw = mcp23s08_port_set_masked_raw,
+	.port_set_bits_raw = mcp23s08_port_set_bits_raw,
+	.port_clear_bits_raw = mcp23s08_port_clear_bits_raw,
+	.port_toggle_bits = mcp23s08_port_toggle_bits,
+	.pin_interrupt_configure = mcp23s08_pin_interrupt_configure,
+};
+
+/**
+ * @brief Initialization function of MCP23S08
+ *
+ * @param dev Device struct
+ * @return 0 if successful, failed otherwise.
+ */
+static int mcp23s08_init(const struct device *dev)
+{
+	const struct mcp23s08_config *const config =
+		dev->config;
+	struct mcp23s08_drv_data *const drv_data =
+		(struct mcp23s08_drv_data *const)dev->data;
+
+	drv_data->spi = device_get_binding((char *)config->spi_dev_name);
+	if (!drv_data->spi) {
+		LOG_DBG("Unable to get SPI device");
+		return -ENODEV;
+	}
+
+	if (config->cs_dev) {
+		/* handle SPI CS thru GPIO if it is the case */
+		drv_data->mcp23s08_cs_ctrl.gpio_dev =
+			device_get_binding(config->cs_dev);
+		if (!drv_data->mcp23s08_cs_ctrl.gpio_dev) {
+			LOG_ERR("Unable to get GPIO SPI CS device");
+			return -ENODEV;
+		}
+
+		drv_data->mcp23s08_cs_ctrl.gpio_pin = config->cs_pin;
+		drv_data->mcp23s08_cs_ctrl.gpio_dt_flags = config->cs_flags;
+		drv_data->mcp23s08_cs_ctrl.delay = 0;
+
+		drv_data->spi_cfg.cs = &drv_data->mcp23s08_cs_ctrl;
+
+		LOG_DBG("SPI GPIO CS configured on %s:%u",
+			config->cs_dev, config->cs_pin);
+	}
+
+	drv_data->spi_cfg.frequency = config->freq;
+	drv_data->spi_cfg.operation = (SPI_OP_MODE_MASTER | SPI_MODE_CPOL |
+				       SPI_MODE_CPHA | SPI_WORD_SET(8) |
+				       SPI_LINES_SINGLE);
+	drv_data->spi_cfg.slave = config->slave;
+
+	k_sem_init(&drv_data->lock, 1, 1);
+
+	return 0;
+}
+
+#define MCP23S08_INIT(inst)						\
+	static struct mcp23s08_config mcp23s08_##inst##_config = {	\
+		.common = {						\
+			.port_pin_mask =				\
+			GPIO_PORT_PIN_MASK_FROM_DT_INST(inst),		\
+		},							\
+		.spi_dev_name = DT_INST_BUS_LABEL(inst),		\
+		.slave = DT_INST_REG_ADDR(inst),			\
+		.freq = DT_INST_PROP(inst, spi_max_frequency),		\
```

Comment:
please update driver to use `spi_dt_spec` (see for example https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/gpio/gpio_mcp23s17.c)

Reply:
Done as part of the more generic work. Thanks!